core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/convert.kt

```diff
@@ -97,66 +280,255 @@ public inline fun <T, C, reified R> Convert<T, C?>.notNull(
+ *
+ * See [Grammar][ConvertDocs.Grammar] for more details.
+ */
 @HasSchema(schemaArg = 0)
 public class Convert<T, out C>(
     @PublishedApi internal val df: DataFrame<T>,
     @PublishedApi internal val columns: ColumnsSelector<T, C>,
 ) {
+    /**
+     * Casts the type parameter of the columns previously selected with [convert][convert] to a new type [R],
+     * without performing any actual data transformation.
+     *
+     * This operation updates the static type of the selected columns for further type-safe conversions.
+     */
     public fun <R> cast(): Convert<T, R> = Convert(df, columns as ColumnsSelector<T, R>)
 
+    /**
+     * Converts values in the columns previously selected with [convert] to the specified type [D],
+     * preserving their original names and positions within the [DataFrame].
+     *
+     * The target type is provided as a reified type argument.
+     * For the full list of supported types,
+     * refer to the [documentation website]({@include [DocumentationUrls.Url]}/convert.html).
+     *
+     * For more information: {@include [DocumentationUrls.Convert]}
+     *
+     * ### Examples:
+     * ```kotlin
+     * // Convert selected columns to Int:
+     * df.convert("year", "count").to<Int>()
+     *
+     * // Convert all String columns to LocalDate:
+     * df.convert { colsOf<String>() }.to<LocalDate>()
+     * ```
+     *
+     * @param D The target type, provided as a reified type argument, to convert values to.
+     * @return A new [DataFrame] with the values converted to type [D].
+     */
     @Refine
     @Interpretable("To0")
     public inline fun <reified D> to(): DataFrame<T> = to(typeOf<D>())
 
     override fun toString(): String = "Convert(df=$df, columns=$columns)"
 }
 
-public fun <T> Convert<T, *>.to(type: KType): DataFrame<T> = to { it.convertTo(type) }
+/**
+ * Converts values in the columns previously selected with [convert] to the specified [type],
+ * preserving their original names and positions within the [DataFrame].
+ *
+ * The target type is provided as a [KType].
+ * For the full list of supported types,
+ * refer to the [documentation website]({@include [DocumentationUrls.Url]}/convert.html).
+ *
+ * For more information: {@include [DocumentationUrls.Convert]}
+ *
+ * ### Examples:
+ * ```kotlin
+ * // Convert selected columns to String:
+ * df.convert("year", "count").to(typeOf<String>())
+ * df.convert { year and count }.to(typeOf<String>())
+ * // Convert all `Int` columns to `Double`
+ * df.convert { colsOf<Int>() }.to(typeOf<Double>())
+ * ```
+ *
+ * @param type The target type, provided as a [KType], to convert values to.
+ * @return A new [DataFrame] with the values converted to [type].
+ */
+public fun <T> Convert<T, *>.to(type: KType): DataFrame<T> = asColumn { it.convertTo(type) }
 
-public fun <T, C> Convert<T, C>.to(columnConverter: DataFrame<T>.(DataColumn<C>) -> AnyBaseCol): DataFrame<T> =
```

Comment:
We should probably add a `@Deprecated` annotation here, it can be ERROR right away but we should steer people to `convert.asColumn`